api_v3/lib/KalturaEntryService.php

```diff
@@ -685,9 +685,11 @@ protected function attachOperationResource(kOperationResource $resource, entry $
 		}
 		
 		$errDescription = '';
-		kBusinessPreConvertDL::decideAddEntryFlavor(null, $dbEntry->getId(), $resource->getAssetParamsId(), $errDescription, $dbAsset->getId(), $operationAttributes);
-		
-		if($isNewAsset)
+		$batchJob = kBusinessPreConvertDL::decideAddEntryFlavor(null, $dbEntry->getId(), $resource->getAssetParamsId(), $errDescription, $dbAsset->getId(), $operationAttributes);
+		$isImportNeeded = false;
+		if ($batchJob && $batchJob->getJobType() == BatchJobType::IMPORT)
+			$isImportNeeded = true;
+		if($isNewAsset && !$isImportNeeded)
```

Comment:
Since the decision is done inside decideAddEntryFlavor, it more sense to check it return value than to add another logic.